controllers/system/system_controller.go

```diff
@@ -84,51 +85,38 @@ func InstallCertificate(filename string, data []byte) error {
 	return nil
 }
 
-// installRootCertificates examines the list of certificates to be configured
-// against the system and for any platform certificates found it will install
-// the corresponding CA certificate into the system certificate path.  Since
-// all controllers need to spawn clients to communicate with the system API
-// this approach was used rather than to pass the certificates directly to the
-// gophercloud API since that would require several additional steps by each
-// controller to load the certificates from the system secret.
+// installRootCertificates installs RestAPI/GUI RCA certificate into the system
+// certificate path. Since all controllers need to spawn clients to communicate
+// with the system API this approach was used rather than to pass the
+// certificates directly to the gophercloud API since that would require several
+// additional steps by each controller to load the certificates from the system
+// secret.
 func (r *SystemReconciler) installRootCertificates(instance *starlingxv1.System) error {
-	if instance.Spec.Certificates == nil {
-		return nil
+	secret := v1.Secret{}
+	secretName := types.NamespacedName{Namespace: instance.Namespace, Name: RestAPIcertName}
+	err := r.Client.Get(context.TODO(), secretName, &secret)
+	if err != nil {
+		return err
 	}
 
-	for _, c := range *instance.Spec.Certificates {
-		if c.Type != starlingxv1.PlatformCertificate {
-			// We only interact with the platform API therefore we do not need
-			// to install any other CA certificate locally.
-			continue
-		}
-
-		secret := v1.Secret{}
-		secretName := types.NamespacedName{Namespace: instance.Namespace, Name: c.Secret}
-		err := r.Client.Get(context.TODO(), secretName, &secret)
-		if err != nil {
-			return err
-		}
-
-		var caBytes []byte
-		var ok bool
-		numRetries := 30
-		for iter := 0; iter < numRetries; iter++ {
-			caBytes, ok = secret.Data[starlingxv1.SecretCaCertKey]
-			if !ok {
-				logSystem.Info("Platform certificate CA not ready/available", "name", c.Secret)
-				time.Sleep(5 * time.Second)
-			} else {
-				logSystem.Info("Platform certificate CA ready!")
-				break
-			}
-		}
+	var caBytes []byte
+	var ok bool
+	numRetries := 30
+	for iter := 0; iter < numRetries; iter++ {
+		caBytes, ok = secret.Data[starlingxv1.SecretCaCertKey]
 		if !ok {
-			logSystem.Info("Continuing deployment without a CA certificate", "name", c.Secret)
-			continue
+			logSystem.Info("Platform certificate CA not ready/available", "name", RestAPIcertName)
+			time.Sleep(5 * time.Second)
```

Comment:
Is it okay if we just wait for 5s and move forward? Do we need to have a retry?

Reply:
Previously we would do it for every cert in the spec, se we would run it again for the next crt.
I didn't see a failure on this in my tests, so I'm not sure. If you think it would be good, I can add retries (maybe 3?).

Reply:
No, I see it is from the previous code and in the loop, sorry